lib/msf/core/post/linux/kernel.rb

```diff
@@ -0,0 +1,69 @@
+# -*- coding: binary -*-
+require 'msf/core/post/common'
+
+module Msf
+class Post
+module Linux
+module Kernel
+  include ::Msf::Post::Common
+
+  #
+  # Returns the kernel version
+  #
+  # @return [String]
+  #
+  def kernel_version
+    cmd_exec('uname -r').to_s
+  rescue
+    raise 'Could not determine kernel version'
+  end
+
+  #
+  # Returns the kernel modules
+  #
+  # @return [String]
+  #
+  def kernel_modules
+    cmd_exec('cat /proc/modules').to_s
+  rescue
+    raise 'Could not determine kernel modules'
+  end
+
+  #
+  # Returns true if kernel and hardware supports Supervisor Mode Access Prevention (SMAP), false if not.
+  #
+  # @return [Boolean]
+  #
+  def smap_enabled?
+    cmd_exec('cat /proc/cpuinfo').to_s.include? 'smap'
+  rescue
+    raise 'Could not determine SMAP status'
+  end
+
+  #
+  # Returns true if kernel and hardware supports Supervisor Mode Execution Protection (SMEP), false if not.
+  #
+  # @return [Boolean]
+  #
+  def smep_enabled?
+    cmd_exec('cat /proc/cpuinfo').to_s.include? 'smep'
+  rescue
+    raise 'Could not determine SMEP status'
+  end
+
+  #
+  # Returns true if user namespaces are enabled, false if not.
+  #
+  # @return [Boolean]
+  #
+  def userns_enabled?
+    return false if cmd_exec('cat /proc/sys/user/max_user_namespaces').to_s.eql? '0'
```

Comment:
why `eql?`?  It isn't incorrect, but I would say that `==` is preferred in most styles.  Maybe?

Reply:
It's my style. It's not the preferred style.

On the other hand, it's impossible to accidentally to do this with `.eql?` :

```ruby
return false if cmd_exec('cat /proc/sys/user/max_user_namespaces').to_s = '0'
```

Instead of this:

```ruby
return false if cmd_exec('cat /proc/sys/user/max_user_namespaces').to_s == '0'
```

Reply:
Good point.  👍